### ISSUE TITLE
error initializing provisioner 'shell' on packer version 1.6.2 and 1.6.3

### ISSUE THREAD

**Issue by ture-karlsson** [hashicorp/packer#10005]
After upgrading packer from version 1.6.0 to 1.6.2, we started experiencing the following kind of errors in our builds:

    error initializing provisioner 'shell': EOF

and

    error initializing provisioner 'shell': timeout while waiting for plugin to start

We experience the same error after upgrading to 1.6.3, but when downgrading to 1.6.0, we can run our builds again without issues. 

Are there any changes in versions later than 1.6.0 that can cause these issues?

**Comment by sylviamoss** [comment]
Hey there, thanks for reaching out! Could you provide us more information on what builder you're using and a simplificated template and steps we can use to reproduce the issue?

**Comment by ture-karlsson** [comment]
Hello! 

We are using the google compute builder and running it in cloud build. We are only experiencing this with the shell provisioner and don't use any custom plugins. 

Simplified template looks something like this:

    {
      "builders": [
        {
          "type": "googlecompute",
          "name": "example-name",
          "project_id": "example-project",
          "image_name": "example-image",
          ...
        }
      ],
      "provisioners": [
        {
          "type": "shell",
          "script": "path/to/script.sh"
        }
    }

I will try to come up with a reproducer.

**Comment by ture-karlsson** [comment]
Hello!

I have tried to reproduce the error without running our full build, but with no success. As I said, this is in GCP Cloud Build and we have 20+ steps running in parallel and the errors does not appear on the same step every time and does not appear when I run any of the steps individually, nor if I create a simple mock template like the one above and run it in 30 parallel steps.

Do you have any suggestions on how we can narrow this down?

**Comment by SwampDragons** [comment]
Hi, it sounds to me like you're hitting some kind of compute limit of the instance you're building on. Try splitting your Packer run across multiple cloud builds, or running on a more powerful instance. Here's some documentation about available instance types: https://cloud.google.com/compute/docs/machine-types#n1_machine_type

**Comment by ture-karlsson** [comment]
Yes, I think you are correct! Running on a bigger machine worked fine:

`gcloud builds submit --config=config.yaml ---machine-type=n1-highcpu-32`

Are there any changes between 1.6.0 and 1.6.2/1.6.3 that would make the builds require more resources?

**Comment by SwampDragons** [comment]
I'm definitely going to look into that based on this report.

**Comment by tanmay-hc** [comment]
This seems like an old issue which might not be relevant anymore since we are now at v1.12.0.
Please inform us if any such problem occurs again.
Closing this for now. Feel free to create a new issue if you face any issue.
